contrib/install_db4.sh

```diff
@@ -63,8 +63,12 @@ http_get() {
 }
 
 mkdir -p "${BDB_PREFIX}"
-http_get "${BDB_URL}" "${BDB_VERSION}.tar.gz" "${BDB_HASH}"
-tar -xzvf ${BDB_VERSION}.tar.gz -C "$BDB_PREFIX"
+if [ -f "${BDB_VERSION}.tar.gz" ]; then
```

Comment:
For the current logic it would imo be better to write it as a negation so that the `tar` can be factored out:
```bash
if [ ! -f "${BDB_VERSION}.tar.gz" ]; then
    http_get "${BDB_URL}" "${BDB_VERSION}.tar.gz" "${BDB_HASH}"
fi
tar -xzvf ${BDB_VERSION}.tar.gz -C "$BDB_PREFIX"
```
I agree that checking the hash would make sense even if the file exists, though.

Reply:
> For the current logic it would imo be better to write it as a negation so that the `tar` can be factored out:
> 
> ```shell
> if [ ! -f "${BDB_VERSION}.tar.gz" ]; then
>     http_get "${BDB_URL}" "${BDB_VERSION}.tar.gz" "${BDB_HASH}"
> fi
> tar -xzvf ${BDB_VERSION}.tar.gz -C "$BDB_PREFIX"
> ```

That suggestion is good if http_get exits upon failure.

You are not proposing any error control on the tar exit status,
are you? It would fail if http_get fails what is supposed to do.

Reply:
changed to `if not exist` logic